src/components/BSelect/BSelect.vue

```diff
@@ -0,0 +1,284 @@
+<template>
+  <div>
+    <div>
+      <label>{{ selectLabel }} <span v-if=isRequired aria-label='Required'>*</span></label>
+    </div>
+    <div 
+      :aria-expanded="opened"
+      :aria-label="selectLabel"
+      :aria-owns="id"
+      :aria-required="isRequired"
+      :id="'dropdown' + id"
+      @keyup.32.prevent.stop="toggleList"
+      role="combobox"
+      tabindex="0"
+      v-model="selectLabel"
+    >
+      <ul
+        :class="ulClass"
+        :id="id"
+        role="listbox"
+      >
+        <li 
+          :class="dropdownClass" 
+          @click="toggleList"
+          >{{ selectedOption ? selectedOption.text : placeholder }}
+          <span class="dropdown-arrow"><i class="fa fa-caret-down" aria-hidden="true"></i></span>
+        </li>
+        <span v-show="selectErrors.has('selectedOption')" class="error" data-vv-as="selectLabel">{{ selectErrors.first('selectedOption') }}</span>
+        <li
+          v-if=opened
+          v-for="option in options"
+          :aria-selected="selectedOption && selectedOption.id === option.id"
+          :id="'option' + option.id"
+          :key="'selectOption' + option.id"
+          :value=option.id
+          @click="selectOption(option)"
+          @keyup.38.prevent="upHandler"
+          @keyup.40.prevent="downHandler"
+          @keyup.13.prevent="selectOption(option)"
+          @keyup.32.prevent.stop="selectOption(option)"
```

Comment:
You can actually use things like `@keyup.up` https://vuejs.org/v2/guide/events.html#Key-Modifiers

Reply:
im switching it, just had a conversation with normal about this